client/components/date-range/header.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent } from 'react';
+import { noop } from 'lodash';
+import { localize, LocalizeProps } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import SharedProps from './shared-props';
+
+interface Props {
+	onApplyClick: () => void;
+	onCancelClick: () => void;
+	applyButtonText: string | null | undefined;
+	cancelButtonText: string | null | undefined;
+}
+
+export const DateRangeHeader: FunctionComponent< Props & SharedProps & LocalizeProps > = props => {
```

Comment:
As a simple function component, how about `useTranslate` instead of the `localize` wrapper?

```suggestion
export default DateRangeHeader: FunctionComponent< Props & SharedProps > = props => {
  const translate = useTranslate();
```

Reply:
Done!